down_frab_videos.py

```diff
@@ -615,8 +624,12 @@ def __get_fahrplan_as_text(self, fahrplan_json):
             # Request body as a unicode string
             return req.text
 
-    def __init__(self, fahrplan_page):
-        self.__location = fahrplan_page + "/schedule.json"
+    def __init__(self, fahrplan_page, is_json):
```

Comment:
Here I would instead have a constructor with two arguments, one being the `base_page` and one being the `json_location`. The latter would have a default of `None` in which case it was set to `base_page + "/schedule.json"` as it is now.

This has the advantage that `self.base_page` is still meaningful, which is important, since this is used to generate some urls for the attachments. I am not sure if this would still work in the case of GPN?

Reply:
That sounds reasonable. Could you do that? I would have to look up a couple of things. I rarely write Python.

Reply:
Will do ... have little time right now, however.

Reply:
Ok. Who ever gets there first. :D